app/src/main/java/io/homeassistant/companion/android/notifications/MessagingService.kt

```diff
@@ -182,17 +238,112 @@ class MessagingService : FirebaseMessagingService() {
     ) {
 
         val colorString = data["color"]
-        var color = ContextCompat.getColor(this, R.color.colorPrimary)
+        val color = parseColor(colorString, R.color.colorPrimary)
+        builder.color = color
+    }
 
+    private fun parseColor(colorString: String?, default: Int): Int {
         if (!colorString.isNullOrBlank()) {
             try {
-                color = Color.parseColor(colorString)
+                return Color.parseColor(colorString)
             } catch (e: Exception) {
                 Log.e(TAG, "Unable to parse color", e)
             }
         }
+        return ContextCompat.getColor(this, default)
+    }
 
-        builder.color = color
+    private fun handleLegacyLedColor(
+        builder: NotificationCompat.Builder,
+        data: Map<String, String>
+    ) {
+        val ledColor = data[LED_COLOR]
+        if (!ledColor.isNullOrBlank()) {
+            builder.setLights(parseColor(ledColor, R.color.colorPrimary), 3000, 3000)
+        }
+    }
+
+    private fun handleLegacyVibrationPattern(
```

Comment:
So I think that we process vibrations like this right now because it aligns both iOS and Android notifications.

Reply:
It's android legacy not pattern legacy.  Since Android Oreo a channel is needed and that's where you set the vibrate pattern instead of pre Oreo's on the notification itself.

Reply:
So this should be ok i guess?